src/sagas/SignIn.ts

```diff
@@ -0,0 +1,43 @@
+import { Frost } from '@poetapp/frost-client'
+import { Actions } from 'actions/index'
+import { browserHistory } from 'react-router'
+import { delay, SagaIterator } from 'redux-saga'
+import { call, takeLatest, put, ForkEffect } from 'redux-saga/effects'
+const { toast } = require('react-toastify')
+
+export async function signInFrost(data: {
+  readonly email: string
+  readonly password: string
+}): Promise<{ readonly token: string }> {
+  const { email, password } = data
+  const frost = new Frost({ host: '/api' })
+  return await frost.login(email, password)
+}
+
+export function SignInSaga(): () => IterableIterator<ForkEffect> {
+  return function*(): IterableIterator<ForkEffect> {
+    yield takeLatest(Actions.SignIn.SIGN_IN, SignIn)
+  }
+}
+
+export function* SignIn(action: any): SagaIterator {
+  try {
+    const { email, password } = action.payload
+    yield put(Actions.LoadingPage.onLoadingOn())
+    const { token } = yield call(signInFrost, { email, password })
+    yield put(Actions.SignIn.onSignInSuccess({ token, ...{ profile: { email } } }))
```

Comment:
What's the advantage of creating an object and immediately spreading it?

Isn't `{ token, ...{ profile: { email } } }` the same as `{ token, profile: { email } }`?